riscv-pages/src/sequential_pages.rs

```diff
@@ -236,6 +264,43 @@ impl<S: State, I: Iterator<Item = Page<S>>> fmt::Debug for Error<S, I> {
     }
 }
 
+pub struct SeqPageChunkIter<S: State> {
+    pages: SequentialPages<S>,
+    n: u64,
+}
+
+impl<S: State> Iterator for SeqPageChunkIter<S> {
+    type Item = SequentialPages<S>;
+
+    fn next(&mut self) -> Option<SequentialPages<S>> {
+        if self.pages.count == 0 {
+            return None;
+        }
+        let addr = self.pages.addr;
+        let page_size = self.pages.page_size;
+        let count = if self.pages.count > self.n {
+            self.n
+        } else {
+            self.pages.count
+        };
+        self.pages.addr = self
+            .pages
+            .addr
+            .checked_add_pages_with_size(count, page_size)
+            .unwrap();
+        self.pages.count -= count;
+        // Safe because `pages` owns the memory, and we're adding to this sequence pages that have been removed.
+        unsafe { Some(SequentialPages::from_mem_range(addr, page_size, count).unwrap()) }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let count = self.pages.count as usize;
+        (count, Some(count))
+    }
```

Comment:
Shouldn't `size_hint()` return the number of chunks remaining in the iterator, rather than the number of pages?

Reply:
Yes.